specs/agents/metadata.md

```diff
@@ -25,41 +25,34 @@ System metadata relates to the host/container in which the service being monitor
 #### Hostname
 
 This hostname reported by the agent is mapped by the APM Server to the 
-[`host.hostname` ECS field](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-hostname), which should 
-typically contain what the `hostname` command returns on the host machine. However, since we rely on this field for 
-our integration with beats data, we should attempt to follow a similar logic to the `os.Hostname()` Go API, which beats 
-relies one. While `os.Hostname()` contains some complex OS-specific logic to cover all sorts of edge cases, our 
-algorithm should be simpler. It relies on the execution of external commands with a fallback to standard environment 
-variables. Agents SHOULD implement this hostname discovery algorithm wherever possible:
+[`host.hostname` ECS field](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-hostname).
+
+Agents SHOULD return the lower-cased FQDN whenever possible, which might require a DNS query.
+
+Agents SHOULD implement this hostname discovery algorithm wherever possible:
 ```
 var hostname;
 if os == windows
-  hostname = exec "cmd /c hostname"                   // or any equivalent *
+  hostname = exec "powershell.exe [System.Net.Dns]::GetHostEntry($env:computerName).HostName" // or any equivalent *
```

Comment:
[for reviewer] using powershell is not ideal and a better API should be privileged if available, but at least that provides a way to implement it consistently across agents if no such API is available.
I've tested this with a windows machine and a configured custom DNS suffix and it returned the FQDN as expected, however when used with a proper AD domain I'm just assuming that it would work as well :crossed_fingers: .

Reply:
Sounds good to me. References: https://stackoverflow.com/questions/12268885/powershell-get-fqdn-hostname 
https://learn.microsoft.com/en-us/dotnet/api/system.net.dns.gethostentry